### ISSUE TITLE
Files sometimes returns to root folder

### ISSUE THREAD

**Issue by SiRo24** [files-community/Files#6881]
### Description

Sometimes, when I re-enter an opened tab or want to one go one step backwards, Files jumps back to the nearest parent folder pinned to the sidebar.
Let's say you are in a folder which is not a subfolder of any pinned folder, it jumps to the drive root like C:\
If a parent folder is available, it jumps to that
If multiple parent folders are available, it jumps to the closest one.

This isn't a new issue, I experienced it for the last couple of weeks or months?

### Expected behavior

Showing the folder I selected.

### Files Version

v2.0.35.0

### Windows Version

Windows 11 Home Insider Preview 22499.1000

**Comment by gave92** [comment]
Thanks for the report. Strange behavior. Could you share a video recording of this?

**Comment by SiRo24** [comment]
https://youtu.be/I6r23pHebLU

did a small Video, hope it helps

**Comment by yaichenbaum** [comment]
@SiRo24 is this still an issue in the preview version? http://files.community/download/preview

**Comment by SiRo24** [comment]
@yair100 haven't experienced it in a while. So the issue could have been fixed. 
But like I wrote, this issue didn't occure regularly, so maybe it's still there and I just didn't experience it (haven't used Files too much the last few weeks because "replace default file explorer" didn't work (it works now)).
But I think this issue can be closed for now, I can open another one if i notice this bug again.
